### PR TITLE
Created an empty pygame project.

### PR DESCRIPTION
**Pull request by nzevgolisda**

### CODE REVIEW

**Comment by createyourpersonalaccount** [comment]
This should be in a function called `runGame` that takes as argument a configuration class that contains the window size. Use `clock.tick()` to set the framerate to 60 fps.

See the pygame example https://github.com/pygame/pygame/blob/main/examples/chimp.py as well.

File: src/minesweeper/Config.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+import pygame
+from ctypes.wintypes import RGB
```
Comment (createyourpersonalaccount) [inline src/minesweeper/Config.py]: Don't use the `ctypes.wintypes` module, it is useful for calling C code from python, but we won't do that. Most of the code should be coming from pygame.

Instead, use pygame's `pygame.Color`.

File: src/minesweeper/Config.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (createyourpersonalaccount) [inline src/minesweeper/Config.py]: Add the GPL license header here, include both my name and yours.

File: src/minesweeper/Config.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+import pygame
+from ctypes.wintypes import RGB
+
+class Config:
+    def __init__(self):
+        self.screen_width = 120
+        self.screen_height = 80
+        self.bg_color = RGB(0, 0, 0)
```
Comment (createyourpersonalaccount) [inline src/minesweeper/Config.py]: Replace with `pygame.Color("black")`.

File: src/minesweeper/minesweeper.py [hunk 4]
```diff
@@ -12,3 +12,30 @@
...
+
+import sys
+import pygame
+from Config import Config
+from ctypes.wintypes import RGB
+
+def runGame():
+    '''This should be in a function called runGame that takes as argument a configuration class 
+    that contains the window size. Use clock.tick() to set the framerate to 60 fps.
+    See the pygame example https://github.com/pygame/pygame/blob/main/examples/chimp.py as well.'''
```
Comment (createyourpersonalaccount) [inline src/minesweeper/minesweeper.py]: This is called the "docstring". Here you should have the description of `runGame`. We will later use [sphinx](https://www.sphinx-doc.org/en/master/) to automatically extract those into HTML documentation, so use the format

```python
"""Brief description"""
```

or the format

```python
"""
Brief description

Long multi-line description of
everything.

With paragraphs and _Markdown_.
"""
```

File: src/minesweeper/minesweeper.py [hunk 5]
```diff
@@ -12,3 +12,30 @@
...
+
+    pygame.init()
```
Comment (createyourpersonalaccount) [inline src/minesweeper/minesweeper.py]: `runGame` should not be initializing the pygame module, this should be happening in `__init__.py`.

File: src/minesweeper/minesweeper.py [hunk 6]
```diff
@@ -12,3 +12,30 @@
...
+        #Watch for keyboard and mouse events.
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                sys.exit()
+        #Redraw the screen during each pass through the loop.
+        screen.fill(config.bg_color)
+        #Make the most recently drawn screen visible
+        pygame.display.flip()
+        print(config.clock.tick(1))
+runGame()
```
Comment (createyourpersonalaccount) [inline src/minesweeper/minesweeper.py]: Don't call functions in modules. I think you left that in there accidentally when you committed. We will soon add the facilities that will permit us to test the code we write as we write it.

File: src/minesweeper/Config.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+        self.clock = pygame.time.Clock()
```
Comment (createyourpersonalaccount) [inline src/minesweeper/Config.py]: The configuration should not have runnable code like `Clock()`. Instead, save the FPS as `60`, for example,

```python
self.FPS = 60
```

and use `clock = pygame.time.Clock()` in `runGame` together with `clock.tick(config.FPS)`.

File: src/minesweeper/minesweeper.py [hunk 8]
```diff
@@ -12,3 +12,30 @@
...
+from Config import Config
+from ctypes.wintypes import RGB
+
+def runGame():
+    '''This should be in a function called runGame that takes as argument a configuration class 
+    that contains the window size. Use clock.tick() to set the framerate to 60 fps.
+    See the pygame example https://github.com/pygame/pygame/blob/main/examples/chimp.py as well.'''
+
+    pygame.init()
+    pygame.display.set_caption("Minesweeper")
```
Comment (createyourpersonalaccount) [inline src/minesweeper/minesweeper.py]: The string `"Minesweeper"` should be part of the configuration.

File: src/minesweeper/minesweeper.py [hunk 9]
```diff
@@ -12,3 +12,30 @@
...
+    config = Config()
```
Comment (createyourpersonalaccount) [inline src/minesweeper/minesweeper.py]: `config` should be an argument of `runGame`, and the initialization should happen at `__init__.py`.

File: src/minesweeper/minesweeper.py [hunk 10]
```diff
@@ -12,3 +12,30 @@
...
+    while True:
+        #Watch for keyboard and mouse events.
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                sys.exit()
+        #Redraw the screen during each pass through the loop.
+        screen.fill(config.bg_color)
+        #Make the most recently drawn screen visible
+        pygame.display.flip()
+        print(config.clock.tick(1))
```
Comment (createyourpersonalaccount) [inline src/minesweeper/minesweeper.py]: Don't print the result, and change the line as explained above.

File: src/minesweeper/minesweeper.py [hunk 11]
```diff
@@ -12,3 +12,30 @@
...
+    pygame.display.set_caption("Minesweeper")
+    config = Config()
+    screen = pygame.display.set_mode((config.screen_width, config.screen_height))
+    #Start the main loop for the game.
+    while True:
+        #Watch for keyboard and mouse events.
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                sys.exit()
+        #Redraw the screen during each pass through the loop.
```
Comment (createyourpersonalaccount) [inline src/minesweeper/minesweeper.py]: This comment is not necessary, please remove.

File: src/minesweeper/minesweeper.py [hunk 12]
```diff
@@ -12,3 +12,30 @@
...
+        screen.fill(config.bg_color)
+        #Make the most recently drawn screen visible
```
Comment (createyourpersonalaccount) [inline src/minesweeper/minesweeper.py]: This comment is not necessary, please remove.

File: src/minesweeper/Config.py [hunk 13]
```diff
@@ -0,0 +1,10 @@
+
+import pygame
+from ctypes.wintypes import RGB
+
+class Config:
```
Comment (createyourpersonalaccount) [inline src/minesweeper/Config.py:3]: Rename the class to `gameConfig` to distinguish it from other potential configurations there might be.

**Review by createyourpersonalaccount — CHANGES_REQUESTED** [review]
Please make the following changes as noted. Also, please rename the file `Config.py` to `config.py` (python modules [should only have lowercase and underscores](https://www.python.org/dev/peps/pep-0008/#package-and-module-names)), and also add your name in the license header of `minesweeper.py`. It should look like

```
# Copyright (C) 2022  Nikolaos Chatzikonstantinou <nchatz314@gmail.com>,
#                     Nikolaos Zevgolis <nzevgolisda@gmail.com>
```

on all the files that you have modified, and also on new files you create (my name is added since I am giving all the comments and reviews). This is not strictly necessary as we have already added ourselves to the `LICENSE.txt` file, but please do it.

File: src/minesweeper/minesweeper.py (hunk 12 above)
Comment (nzevgolisda) [reply src/minesweeper/minesweeper.py]: Indeed.

**Pull request closed without merging**